readthedocs/builds/models.py

```diff
@@ -291,6 +294,19 @@ def vcs_url(self):
 
     @property
     def last_build(self):
+        # TODO deprecated in favor of `latest_build`, which matches naming on
+        # the Project model
+        return self.latest_build
+
+    @property
+    def latest_build(self):
+        # Check if there is `_latest_build` prefetch in the Queryset.
+        # Used for database optimization.
+        if hasattr(self, self.LATEST_BUILD_CACHE):
+            if latest_build := getattr(self, self.LATEST_BUILD_CACHE):
```

Comment:
I don't think we need the first `if` here?

Reply:
I started to write it like that but we do need to test for `hasattr` first, as this attribute only exists after prefetch and it's what shortcuts this method to avoid the `return self.builds....` below.

If this was just `getattr`, we'd _always_ skip the `return self.builds...` below.